### PR TITLE
Handle daemon updates

### PR DESCRIPTION
**Pull request by JAORMX**
We were not taking into account potential updates to the daemonset's
image. This fixes that by handling that in the reconcile loop.

### CODE REVIEW

File: pkg/controller/fileintegrity/fileintegrity_controller.go [hunk 1]
```diff
@@ -369,47 +366,28 @@ func (r *ReconcileFileIntegrity) Reconcile(request reconcile.Request) (reconcile
...
-				newArgs = append(newArgs, arg)
-			}
-		}
-		newArgs = append(newArgs, fmt.Sprintf("--interval=%s", newGracePeriod))
-		newArgs = append(newArgs, fmt.Sprintf("--debug=%s", newDebug))
-		return true, newArgs, nil
-	}
-	return false, newArgs, nil
+// Returns true with the daemon pod image differs from the current DS.
+// Returns false if there was no difference.
```
Comment (mrogers950) [inline pkg/controller/fileintegrity/fileintegrity_controller.go:382]: note that it also updates the image ref

File: pkg/controller/fileintegrity/fileintegrity_controller.go [hunk 2]
```diff
@@ -369,47 +366,28 @@ func (r *ReconcileFileIntegrity) Reconcile(request reconcile.Request) (reconcile
...
-		if intervalOpt(arg) {
-			currentGracePeriod = arg[len("--interval="):]
-		} else if debugOpt(arg) {
-			currentDebug = arg[len("--debug="):]
-		}
-	}
-	if currentGracePeriod == "" || currentDebug == "" {
-		return false, newArgs, fmt.Errorf("bad daemon configuration")
-	}
+func updateDSArgs(currentDS *appsv1.DaemonSet, fi *fileintegrityv1alpha1.FileIntegrity) bool {
```
Comment (mrogers950) [inline pkg/controller/fileintegrity/fileintegrity_controller.go:371]: update this comment too

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 2a2c375f8abe31b1b56fd8d99c8af73218fe748c

File: pkg/controller/fileintegrity/fileintegrity_controller.go (hunk 2 above)
Comment (JAORMX) [reply pkg/controller/fileintegrity/fileintegrity_controller.go:370]: done, thanks!

File: pkg/controller/fileintegrity/fileintegrity_controller.go (hunk 1 above)
Comment (JAORMX) [reply pkg/controller/fileintegrity/fileintegrity_controller.go:381]: done, thanks!

**Pull request merged**